internal/cmd/local/command_kafka_topic_produce.go

```diff
@@ -0,0 +1,170 @@
+	"github.com/confluentinc/cli/internal/pkg/serdes"
+)
+
+func (c *command) newProduceCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "produce <topic>",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.topicProduce,
+		Short: "Produce messages to a Kafka topic.",
+		Long:  "Produce messages to a Kafka topic. Configuration and command guide: https://docs.confluent.io/confluent-cli/current/cp-produce-consume.html.\n\nWhen using this command, you cannot modify the message header, and the message header will not be printed out.",
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Produce message to topic "my_topic" providing key.`,
+				Code: `confluent local kafka topic produce my_topic --parse-key`,
+			},
+		),
+	}
+
+	cmd.Flags().Bool("parse-key", false, "Parse key from the message.")
+	cmd.Flags().String("delimiter", ":", "The delimiter separating each key and value.")
+	cmd.Flags().StringSlice("config", nil, `A comma-separated list of configuration overrides ("key=value") for the producer client.`)
+	cmd.Flags().String("config-file", "", "The path to the configuration file (in json or avro format) for the producer client.")
+	return cmd
+}
+
+func (c *command) topicProduce(cmd *cobra.Command, args []string) error {
+	if c.Config.LocalPorts == nil {
+		return errors.NewErrorWithSuggestions(errors.FailedToReadPortsErrorMsg, errors.FailedToReadPortsSuggestions)
+	}
+	producer, err := newOnPremProducer(cmd, ":"+c.Config.LocalPorts.PlaintextPort)
+	if err != nil {
+		return errors.NewErrorWithSuggestions(fmt.Errorf(errors.FailedToCreateProducerErrorMsg, err).Error(), errors.OnPremConfigGuideSuggestions)
+	}
+	defer producer.Close()
+	log.CliLogger.Tracef("Create producer succeeded")
+
+	adminClient, err := ckafka.NewAdminClientFromProducer(producer)
+	if err != nil {
+		return fmt.Errorf(errors.FailedToCreateAdminClientErrorMsg, err)
+	}
+	defer adminClient.Close()
+
+	topicName := args[0]
+	err = kafka.ValidateTopic(adminClient, topicName)
+	if err != nil {
+		return err
+	}
+
+	serializationProvider, err := serdes.GetSerializationProvider("string")
+	if err != nil {
+		return err
+	}
+
+	output.ErrPrintln(errors.StartingProducerMsg)
+
+	// Line reader for producer input.
+	scanner := bufio.NewScanner(os.Stdin)
+	// On-prem Kafka messageMaxBytes: using the same value of cloud. TODO: allow larger sizes if customers request
+	// https://github.com/confluentinc/cc-spec-kafka/blob/9f0af828d20e9339aeab6991f32d8355eb3f0776/plugins/kafka/kafka.go#L43.
+	const maxScanTokenSize = 1024*1024*2 + 12
+	scanner.Buffer(nil, maxScanTokenSize)
+	input := make(chan string, 1)
+	// Avoid blocking in for loop so ^C or ^D can exit immediately.
+	var scanErr error
+	scan := func() {
+		hasNext := scanner.Scan()
+		if !hasNext {
+			// Actual error.
+			if scanner.Err() != nil {
+				scanErr = scanner.Err()
+			}
+			// Otherwise just EOF.
+			close(input)
+		} else {
+			input <- scanner.Text()
+		}
+	}
```

Comment:
Any way we can avoid duplicating this code? If we change this functionality in the future, it would be great not to have to change it in two places.